curio/local.py

```diff
@@ -58,22 +58,28 @@
 from contextlib import contextmanager
 
 # The thread-local storage slot that points to the task-local storage dict for
-# whatever task is currently running.
-_current_task_local_storage = threading.local()
-_current_task_local_storage.value = None
+# whatever task is currently running in whichever thread is currently running.
+_thread_local_storages = {}
 
+def _local_storage():
+    thread = threading.current_thread()
+    if thread not in _thread_local_storages:
+        _thread_local_storages[thread] = threading.local()
+        _thread_local_storages[thread].value = None
+    return _thread_local_storages[thread]
```

Comment:
So uh... this is exactly a reimplementation of what `threading.local` does (only `threading.local` is more efficient because it can advantage of the platform native mechanisms).